xchange-blockchain/src/main/java/org/knowm/xchange/blockchain/BlockchainExchange.java

```diff
@@ -1,32 +1,56 @@
 package org.knowm.xchange.blockchain;
 
 import org.knowm.xchange.BaseExchange;
-import org.knowm.xchange.Exchange;
 import org.knowm.xchange.ExchangeSpecification;
+import org.knowm.xchange.blockchain.service.BlockchainAccountService;
+import org.knowm.xchange.client.ExchangeRestProxyBuilder;
+import org.knowm.xchange.client.ResilienceRegistries;
 import si.mazi.rescu.SynchronizedValueFactory;
 
-/** @author Tim Molter */
-public class BlockchainExchange extends BaseExchange implements Exchange {
+import javax.ws.rs.HeaderParam;
+
+import static org.knowm.xchange.blockchain.BlockchainConstants.*;
+
+/** @author scuevas*/
+public class BlockchainExchange extends BaseExchange {
+
+  protected static ResilienceRegistries RESILIENCE_REGISTRIES;
+
+  protected BlockchainAuthenticated blockchain;
 
   @Override
-  public ExchangeSpecification getDefaultExchangeSpecification() {
+  protected void initServices() {
+    this.blockchain = ExchangeRestProxyBuilder
+            .forInterface(BlockchainAuthenticated.class, this.getExchangeSpecification())
+            .clientConfigCustomizer(clientConfig -> {
+              String secretKey = this.getExchangeSpecification().getSecretKey();
+              clientConfig.addDefaultParam(HeaderParam.class, X_API_TOKEN, secretKey);
```

Comment:
In this case it is complicated to move secretKey because BlockchainConstants should extend BaseExchange in order to use getExchangeSpecification(). So I directly do this:

```
this.blockchain = ExchangeRestProxyBuilder
            .forInterface(BlockchainAuthenticated.class, this.getExchangeSpecification())
            .clientConfigCustomizer(clientConfig -> clientConfig.addDefaultParam(
                    HeaderParam.class, X_API_TOKEN,
                    this.getExchangeSpecification().getSecretKey())
            ).build();
```